.github/workflows/ci.yml

```diff
@@ -22,169 +22,307 @@ jobs:
   #   steps:
   #     - name: Clone repository
   #       uses: actions/checkout@v2
-  #     - name: Build repository
+  #     - name: Install dependencies
   #       run: |
-  #         mkdir build
-  #         cd build
-  #         cmake .. -DCMAKE_BUILD_TYPE=Debug -DBUILD_TEST=TRUE -DMEMORY_SANITIZER=TRUE
+  #         apk update
+  #         apk upgrade
+  #         apk add alpine-sdk cmake clang linux-headers perl bash openssl-dev
+  #     - name: Build Repository
+  #       run: | 
+  #         mkdir build && cd build
+  #         cmake .. -DBUILD_STATIC_LIBS=TRUE -DBUILD_TEST=TRUE
   #         make
   #         export AWS_KVS_LOG_LEVEL=3
   #         ulimit -c unlimited -S
-  #         timeout --signal=SIGABRT 60m ./tst/webrtc_client_test
-
-
-  # undefined-behavior-sanitizer:
-  #   runs-on: ubuntu-18.04
-  #   env:
-  #     UBSAN_OPTIONS: halt_on_error=1
-  #   steps:
-  #     - name: Clone repository
-  #       uses: actions/checkout@v2
+  #         ../scripts/check-static-build.sh
+  #         ./tst/webrtc_client_test
   #     - name: Build repository
   #       run: |
-  #         # TODO: Remove the following line. This is only a workaround for enabling IPv6, https://github.com/travis-ci/travis-ci/issues/8891.
   #         sudo sh -c 'echo 0 > /proc/sys/net/ipv6/conf/all/disable_ipv6'
-  #         mkdir build
-  #         cd build
-  #         cmake .. -DCMAKE_BUILD_TYPE=Debug -DBUILD_TEST=TRUE -DUNDEFINED_BEHAVIOR_SANITIZER=TRUE
+  #         mkdir build && cd build
+  #         cmake .. -DBUILD_STATIC_LIBS=TRUE -DBUILD_TEST=TRUE
   #         make
   #         export AWS_KVS_LOG_LEVEL=3
   #         ulimit -c unlimited -S
+  #         ../scripts/check-static-build.sh
   #         timeout --signal=SIGABRT 60m ./tst/webrtc_client_test
-
-  # windows-build:
-  #   runs-on: windows-latest
-  #   env:
-  #     PATH: "/c/Strawberry/perl/site/bin:/c/Strawberry/perl/bin:/c/Strawberry/c/bin:/c/Program Files/NASM:`pwd`/open-source/lib:`pwd`/open-source/bin:$PATH"
+  mbedtls-ubuntu-gcc:
+    runs-on: ubuntu-18.04
+    steps:
+      - name: Clone repository
+        uses: actions/checkout@v2
+      - name: Install deps
+        run: |
+          sudo sh -c 'echo 0 > /proc/sys/net/ipv6/conf/all/disable_ipv6'
+          sudo add-apt-repository -y ppa:ubuntu-toolchain-r/test
+          sudo add-apt-repository 'deb http://archive.ubuntu.com/ubuntu/ trusty main'
+          sudo add-apt-repository 'deb http://archive.ubuntu.com/ubuntu/ trusty universe'
+          sudo apt-get -q update
+          sudo apt-get -y install gcc-4.4
+          sudo apt-get -y install gdb
+      - name: Build repository
+        run: |
+          mkdir build && cd build
+          cmake .. -DBUILD_TEST=TRUE -DUSE_OPENSSL=OFF -DUSE_MBEDTLS=ON
+          make
+          export AWS_KVS_LOG_LEVEL=3
+          ulimit -c unlimited -S
+          timeout --signal=SIGABRT 60m ./tst/webrtc_client_test
+  mbedtls-ubuntu-clang:
+    runs-on: ubuntu-18.04
+    env:
+      CC: clang-7
```

Comment:
Any reason for choosing clang-7? 
Latest ubuntu image with just clang is not working, is it?

Reply:
Clang-7 is what travis uses, some other builds were failing because of this, so I set to clang-7 everywhere for stability